### PR TITLE
Cleaning FeedAdapter - step 2

### PR DESCRIPTION
**Pull request by florina-muntenescu**
Removing DataLoadingSubject from FeedAdapter and moving the logic to the Home and Search ViewModels

### CODE REVIEW

File: app/src/test/java/io/plaidapp/ui/HomeViewModelTest.kt [hunk 1]
```diff
@@ -303,12 +308,40 @@ class HomeViewModelTest {
         assertEquals(inactiveSource.key, source)
     }
 
+    @Test
+    fun dataLoading() {
+        // Given a view model
+        val homeViewModel = createViewModel()
+        Mockito.verify(dataManager).registerCallback(capture(dataLoadingCallback))
```
Comment (nickbutcher) [inline app/src/test/java/io/plaidapp/ui/HomeViewModelTest.kt]: Static import?

File: core/src/main/java/io/plaidapp/core/feed/FeedUiModel.kt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import io.plaidapp.core.data.PlaidItem
+
+/**
+ * UI model for feed data
+ */
+data class FeedUiModel(
+    val items: List<PlaidItem>
+)
+
+data class FeedProgressUiModel(
```
Comment (nickbutcher) [inline core/src/main/java/io/plaidapp/core/feed/FeedUiModel.kt]: Should we consider enabling experimental inline classes?

File: search/src/main/java/io/plaidapp/search/ui/SearchViewModel.kt [hunk 3]
```diff
@@ -31,19 +33,34 @@ import io.plaidapp.search.domain.SearchDataManager
...
+
     private val onDataLoadedCallback: OnDataLoadedCallback<List<PlaidItem>> =
         object : OnDataLoadedCallback<List<PlaidItem>> {
             override fun onDataLoaded(data: List<PlaidItem>) {
-                _searchResults.value = Event(data)
+                _searchResults.value = Event(FeedUiModel(data))
             }
         }
 
+    private val dataLoadingCallbacks = object : DataLoadingSubject.DataLoadingCallbacks {
```
Comment (nickbutcher) [inline search/src/main/java/io/plaidapp/search/ui/SearchViewModel.kt]: Worth factoring this out somewhere?

File: core/src/main/java/io/plaidapp/core/feed/FeedUiModel.kt (hunk 2 above)
Comment (florina-muntenescu) [reply core/src/main/java/io/plaidapp/core/feed/FeedUiModel.kt]: as soon as this is not used in java classes, yes.

File: search/src/main/java/io/plaidapp/search/ui/SearchViewModel.kt (hunk 3 above)
Comment (florina-muntenescu) [reply search/src/main/java/io/plaidapp/search/ui/SearchViewModel.kt]: In this PR I wanted to move things out of the `FeedAdapter`. We shouldn't have such a callback at all in our app so, I expect that we will be able to remove it completely in a future PR.

**Comment by florina-muntenescu** [comment]
@nickbutcher - ready for another 👀

File: app/src/main/java/io/plaidapp/ui/HomeActivity.java [hunk 4]
```diff
@@ -174,6 +174,14 @@ protected void onCreate(Bundle savedInstanceState) {
                     checkEmptyState();
                 });
 
+        viewModel.getFeedProgress().observe(this, feedProgressUiModel -> {
```
Comment (nickbutcher) [inline app/src/main/java/io/plaidapp/ui/HomeActivity.java]: Move trailing lambda out of parens?

Comment (florina-muntenescu) [reply app/src/main/java/io/plaidapp/ui/HomeActivity.java]: Not in Java :)

**Pull request merged**